docs: Improve license attribute guidelines (#18566)
conan-io/conan-center-index: docs/adding_packages/conanfile_attributes.md

### License Attribute

The license attribute is a mandatory field which provides the legal information that summarizes the contents saved in the package. These follow the
[SPDX license](https://spdx.org/licenses/) as a standard. This is for consummers, in particular in the enterprise sector, that do rely on SDPX compliant identifiers so that they can flag this as a custom license text.

* If the library has a license that has a SPDX identifier, use the [short Identifiers](https://spdx.dev/ids/).
* If the library has a license text that does not match a SPDX identifier, including custom wording disclaiming copyright or dedicating the words to the ["public domain"](https://fairuse.stanford.edu/overview/public-domain/welcome/), use the [SPDX License Expressions](https://spdx.github.io/spdx-spec/v2-draft/SPDX-license-expressions/), this can follow:

>   * `LicenseRef-` as a prefix, followed by the name of the file which contains the custom license. See [this example](https://github.com/conan-io/conan-center-index/blob/e604534bbe0ef56bdb1f8513b83404eff02aebc8/recipes/fft/all/conanfile.py#L8).

Reviewer: ```suggestion
  * `LicenseRef-` as a prefix, followed by the name of the library. For example:`LicenseRef-libfoo-public-domain`
```

>   * `LicenseRef-` follow by the library name and/or version

Reviewer: ```suggestion
```
